bin/oneoff/backfill_data/user_cap_state_columns

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env ruby
+
+# This script migrates CAP data from properties to separate indexed columns.
+
+require 'ruby-progressbar'
+require_relative '../../../dashboard/config/environment'
+
+progress_bar = ProgressBar.create(
+  total: nil,
+  title: 'Migrating CAP data',
+  format: '%t[%c]: |%B| %a',
+  unknown_progress_animation_steps: %w[>-- ->- -->]
+)
+
+User.select(:id).where(cap_state: nil).where("properties->>'$.child_account_compliance_state' IS NOT NULL").find_in_batches do |users_batch|
+  User.where(id: users_batch).update_all <<~SQL.strip
```

Comment:
> I'm curious to know what the performance implications would be of extracting and adding an index to just the cap_state column, and leaving the cap_state_date column in the blob. Do we have any idea of approximately how many users we expect to be in a locked-out state at any given time?

@Hamms We don't expect a large number of users to be affected initially, as this will only impact Colorado students under 13 for 7 days. However, the number of locked-out users could increase significantly in the future as we add more US states.

Reply:
https://github.com/code-dot-org/code-dot-org/pull/59962#discussion_r1697584729